src/test/kotlin/grpc/GrpcContextTest.kt

```diff
@@ -0,0 +1,62 @@
+package se.uulm.snowballr.backend.grpc
+
+import io.grpc.Context
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import se.uulm.snowballr.backend.model.SnowballRException
+import java.util.UUID
+
+class GrpcContextTest {
+    @Test
+    fun `getUserIdFromContext should return userId when set`() {
+        val testUserId = UUID.randomUUID()
+        val context = Context.current().withValue(GrpcContext.USER_ID_CONTEXT_KEY, testUserId)
+
+        context.run {
+            val result = GrpcContext.getUserIdFromContext()
+            assertEquals(testUserId, result)
+        }
+    }
+
+    @Test
+    fun `getUserIdFromContext should throw MissingUserId when not set`() {
+        val context = Context.current().fork() // ensure a clean context
+
+        context.run {
+            val exception = assertThrows(SnowballRException.MissingContextException.MissingUserId::class.java) {
+                GrpcContext.getUserIdFromContext()
+            }
+            assertNotNull(exception)
```

Comment:
Is there a reason why we don't use the generic parameter to pass the exception type, and why we assert that the exception is not null?
```suggestion
            assertThrows<MissingContextException.MissingUserId> { GrpcContext.getUserIdFromContext() }
```

Same below.

Reply:
No. This was most likely some autocompletion mistake I did not notice.

Reply:
We still don't have to assert that the exception is not null.

Reply:
Removed it. I don't know why I didn't remove these checks the first time.